src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2047,10 +2046,23 @@ var projectUniqueNamesForBuildIntToUpdate
             {
                 // Set the original packages config if it exists
                 var msbuildProject = nuGetProject as MSBuildNuGetProject;
+                var mightNeedHelperNetstandardPackage = false;
+                var needsNetstandard20Assets = false;
+                NuGetFramework projectTargetFramework = null;
                 if (msbuildProject != null)
                 {
                     nuGetProjectContext.OriginalPackagesConfig =
                         msbuildProject.PackagesConfigNuGetProject?.GetPackagesConfig();
+                    projectTargetFramework = msbuildProject.PackagesConfigNuGetProject?.TargetFramework;
+                    if(projectTargetFramework != null)
+                    {
+                        // This boolean is used to narrow down the condition to determine whether package 
+                        // "NETStandard.Library.NetFramework" needs to be installed to this project to make
+                        // NETStandard2.0 compatible with net461/net462/net47.
+                        mightNeedHelperNetstandardPackage = projectTargetFramework == FrameworkConstants.CommonFrameworks.Net461
```

Comment:
I would expect this to compare the version of the framework instead of setting it to just 3. There could be a 464 that would need this still, right? Or is that not possible?